src/core/src/bootstrap/Bootstrapper.py

```diff
@@ -142,36 +143,46 @@ def basic_environment_health_check(self):
         self.composite_logger.log("Process id: " + str(os.getpid()))
 
         # Ensure sudo works in the environment
-        sudo_check_result = self.check_sudo_status()
+        sudo_check_result = self.check_sudo_status_retry()
         self.composite_logger.log_debug("Sudo status check: " + str(sudo_check_result) + "\n")
 
+    def check_sudo_status_retry(self, raise_if_not_sudo=True):
+        """ retry to invoke sudo check """
+        for attempts in range(1, Constants.MAX_CHECK_SUDO_RETRY_COUNT + 1):
+            try:
+                self.composite_logger.log_debug("Performing sudo status check... This should complete within 10 seconds. [Attempts={0}][MaxAttempts={1}]".format(str(attempts), Constants.MAX_CHECK_SUDO_RETRY_COUNT))
+                if self.check_sudo_status(raise_if_not_sudo=raise_if_not_sudo):
+                    self.composite_logger.log_debug("Sudo Check Successfully [Attempts={0}][MaxAttempts={1}]".format(str(attempts), Constants.MAX_CHECK_SUDO_RETRY_COUNT))
+                    return True
```

Comment:
Ignore - only do the function call inside the try catch

Reply:
we need a successful log and a return true here since retry invoke check_sudo_status or else we'll get null exception on this code: sudo_check_result = self.check_sudo_status_with_retry()

Reply:
You don't need a log.

You just need:
return self.check_sudo_status(raise_if_not_sudo)